src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.Utilities.fs

```diff
@@ -129,32 +92,56 @@ module internal Utilities =
         // 3. We are loaded in-process to some other application ... Eg. try .net
         //    See if the host is dotnet.exe ... from net5.0 on this is fairly unlikely
         // 4. If it's none of the above we are going to have to rely on the path containing the way to find dotnet.exe
-        //
+        let fileExists pathToFile =
+            try
+                if File.Exists(pathToFile) then 
+                    true
+                else
+                    false
+            with | _ -> false
+        // Use the path to search for dotnet.exe
+        let probePathForDotnetHost () =
+            let paths =
+                let p = Environment.GetEnvironmentVariable("PATH")
+                if not(isNull p) then p.Split(';') 
```

Comment:
Does this code path get hit on non-Windows?  You'll need to split on `Path.DirectorySeparatorChar` and `Path.AltDirectorySeparatorChar`.

Reply:
Aren't they the '\' and '/' ?

Reply:
This is about splitting the elements of the path environment variable.

Reply:
Japanese systems have a different character, I think the Yen symbol.

Reply:
Hmm it's colon on linux not semi colon.

Reply:
I make this mistake EVERY SINGLE TIME.

What I should have said is `Path.PathSeparator` which is ';' on Windows and ':' on Unix.